From d10aa1882d560d968dd61fb49ad1eeab5df84079 Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Thu, 22 Jul 2010 17:29:20 -0500
Subject: [PATCH 1/3] omap3beagle: Partial revert "HACK: try to poweron stuff on xM rev A"

In this commit:

  commit 0e0b2c1a7a3212e47e2fb07116f1b31914152f76
  Author: Koen Kooi <koen@dominion.thruhere.net>
  Date:   Tue May 4 17:04:27 2010 +0200

      HACK: try to poweron stuff on xM rev A

There is a change introduced to modify the gpio init value after
setting its mode to output.

This was breaking somehow the camera. The sensor powerup sequence
was returning -EBUSY.

Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 arch/arm/mach-omap2/board-omap3beagle.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/board-omap3beagle.c b/arch/arm/mach-omap2/board-omap3beagle.c
index 9bca9fa..3d05bfa 100644
--- a/arch/arm/mach-omap2/board-omap3beagle.c
+++ b/arch/arm/mach-omap2/board-omap3beagle.c
@@ -387,7 +387,7 @@ static int beagle_twl_gpio_setup(struct device *dev,
 
 		/* TWL4030_GPIO_MAX + 0 == ledA, EHCI nEN_USB_PWR (out, active low) */
 		gpio_request(gpio + TWL4030_GPIO_MAX, "nEN_USB_PWR");
-		gpio_direction_output(gpio + TWL4030_GPIO_MAX, 0);
+		gpio_direction_output(gpio + TWL4030_GPIO_MAX, 1);
 	}
 	else {
 		/* DVI reset GPIO is different between revisions */ 
-- 
1.6.6.1

